### ISSUE TITLE
Will not start, failed to create database

### ISSUE THREAD

**Issue by kihaji** [onagre-launcher/onagre#92]
**Describe the bug**
On a new Archlinux install with onagre 1.1.1-1 from the AUR, it will not launch stating it cannot find the database

**Config** 
No config done, straight out of the "box"

**To Reproduce**
On a fresh install of Archlinux, use Paru to install the latest version of Onagre, and from a terminal prompt try and run

**Expected behavior**
It runs

**Screenshots**
2024-12-30T15:50:59.085519Z  INFO onagre: Starting onagre
2024-12-30T15:50:59.085576Z ERROR onagre::app::style: Failed to parse theme "/home/kihaji/.config/onagre/theme.scss": Failed to open config file
2024-12-30T15:50:59.085581Z  WARN onagre::app::style: Failing back to default theme
thread 'main' panicked at src/db/mod.rs:34:55:
failed to create database: Storage(Io(Os { code: 2, kind: NotFound, message: "No such file or directory" }))
stack backtrace:
   0:     0x5a7570f41437 - <unknown>
   1:     0x5a75709dc773 - <unknown>
   2:     0x5a7570f06df2 - <unknown>
   3:     0x5a7570f42b12 - <unknown>
   4:     0x5a7570f4285b - <unknown>
   5:     0x5a7570f4312c - <unknown>
   6:     0x5a7570f42be8 - <unknown>
   7:     0x5a7570f42b49 - <unknown>
   8:     0x5a7570f42b3c - <unknown>
   9:     0x5a7570787bdf - <unknown>
  10:     0x5a7570787fb5 - <unknown>
  11:     0x5a75708f4aba - <unknown>
  12:     0x5a75707e3f46 - <unknown>
  13:     0x5a757090e1ef - <unknown>
  14:     0x5a757092bf66 - <unknown>
  15:     0x5a75708372a3 - <unknown>
  16:     0x5a75709442fa - <unknown>
  17:     0x7ad1ea551e08 - <unknown>
  18:     0x7ad1ea551ecc - __libc_start_main
  19:     0x5a75707c9b25 - <unknown>
  20:                0x0 - <unknown>

**Window manager (please complete the following information):**
 - Window manager:  Hyprland

**Additional context**
I'm pretty sure I'm an idiot and either do not or have a bad environment variable, so once I figure out what the path should be I can fix it, but my Rust is rusty, so it's going to take a second. But at a minimum the error should print out where it is trying to find the database to make it easier to debug.

**Comment by kihaji** [comment]
As a quick update, still the same issue either with or without XDG_DATA_DIRS set

**Comment by oknozor** [comment]
you need either `$XDG_DATA_HOME` or `$HOME`/.local/share

**Comment by kihaji** [comment]
Yeah, it started to work after I installed Brave, which created the .local/share.

If I am reading the XDG spec correctly, $XDG_DATA_HOME defaults to .local/share when it is empty, so to me that means the expected behavior of Onagre on install is if it is unset, and that directory does not exist, it should create it, or at least ask the user if they want to create it/tell them they need to create it.

It's not a huge thing, but a good Quality of Life thing
